monkey/monkey_island/cc/agent_event_handlers/handle_ping_scan_event.py

```diff
@@ -0,0 +1,60 @@
+from logging import getLogger
+
+from common.agent_events import PingScanEvent
+from monkey_island.cc.models import CommunicationType, Machine
+from monkey_island.cc.repository import (
+    IAgentRepository,
+    IMachineRepository,
+    INodeRepository,
+    RetrievalError,
+    StorageError,
+    UnknownRecordError,
+)
+
+logger = getLogger(__name__)
+
+
+class handle_ping_scan_event:
+    def __init__(
+        self,
+        agent_repository: IAgentRepository,
+        machine_repository: IMachineRepository,
+        node_repository: INodeRepository,
+    ):
+        self._agent_repository = agent_repository
+        self._machine_repository = machine_repository
+        self._node_repository = node_repository
+
+    def __call__(self, event: PingScanEvent):
+        if not event.response_received:
+            return
+
+        try:
+            dest_machine = self._get_destination_machine(event)
+            self._update_destination_machine(dest_machine, event)
+            src_machine = self._get_source_machine(event)
+
+            # Update or create the node
+            self._node_repository.upsert_communication(
+                src_machine.id, dest_machine.id, CommunicationType.SCANNED
+            )
+        except (RetrievalError, StorageError, TypeError, UnknownRecordError):
+            logger.exception("Unable to process ping scan data")
+
+    def _get_destination_machine(self, event: PingScanEvent) -> Machine:
+        try:
+            dest_machines = self._machine_repository.get_machines_by_ip(event.target)
+            return dest_machines[0]
+        except UnknownRecordError:
+            machine = Machine(id=self._machine_repository.get_new_id())
```

Comment:
We need to add a network interface to this machine.
```suggestion
            machine = Machine(id=self._machine_repository.get_new_id())
```